### ISSUE TITLE
How to migrate Identity users from MVC5 to ASP.NET Core

### ISSUE THREAD

**Issue by simeyla** [aspnet/AspNetCore.Docs#9770]
See  [this Stack Overflow](https://stackoverflow.com/a/57074910/2263507) question.

This article doesn't describe how to migrate anything. Migration of 'Identity' implies migration of data, but this article doesn't discuss that at all.

There's a package AspNetCoreCompat intended to help with that, but no documentation for it. https://github.com/aspnet/Identity/issues/2043

Some information about how to migrate the schema would be very useful. There is a similar article for Membership &gt; Identity which contains SQL but this article has none.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: db6772c4-1017-5e86-8045-8849b5dd1ad7
* Version Independent ID: bcb8d6b2-f38a-ac10-d3f2-69d4223bfd72
* Content: [Migrate Authentication and Identity to ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/migration/identity?view=aspnetcore-2.1)
* Content Source: [aspnetcore/migration/identity.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/migration/identity.md)
* Product: **aspnet-core**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
You're right it doesn't have any migration of data. What about [Migrate from ASP.NET Membership authentication to ASP.NET Core 2.0 Identity](https://docs.microsoft.com/en-us/aspnet/core/migration/proper-to-2x/membership-to-core-identity?view=aspnetcore-2.1) that you mentioned?

**Comment by Rick-Anderson** [comment]
Thanks for contacting us.
We don’t have the resources to invest in this area, so we are closing the issue. Should your request generate enough  👍 responses, we’ll reconsider.

**Comment by AlissonRS** [comment]
If you don't want to ask users for resetting their passwords, but want to gradually update password hashes to the new algorithm as they authenticate, I posted what I did in [this Stack Overflow](https://stackoverflow.com/a/57074910/2263507) question.

However If you can ask for password resetting, that's better I think.

**Comment by DanJ210** [comment]
Create the controller and actions? I came here for that reason. I'm using the Identity tutorial and I scaffolded Identity to my project and added the migration to my database. The area, views, model context seem to all be set even though I had an existing context and this scaffold created a second one... The only problem is no controller so the "Register" and "Login" buttons or links just reload the page. They go nowhere, there's no controller.

**Comment by Rick-Anderson** [comment]
@DanJ210 you need to [Scaffold Identity](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-3.1&tabs=visual-studio). It's now a RCL.

**Comment by Rick-Anderson** [comment]
@HaoK  this issue is correct, the article vaguely shows how to migrate the UI to ASP.NET Core 1.x.

I think a better approach is to replace this doc with a new one:

* How to migrate  Identity users data from MVC5 to ASP.NET Core. Recommend require new PW approach.
* Create a new MVC app with Identity, scaffold Identity and copy the Identity code to your migration app.

This topic is low traffic, less than 4K user/month, but it's like 4K important users.
@mkArtakMSFT  please set priority.
